custom_components/average/sensor.py

```diff
@@ -535,6 +545,7 @@ async def _async_update_state(
                         last_elapsed = end_ts - last_time
                         value += last_state * last_elapsed
                         elapsed += last_elapsed
+                        current_values.append(last_state)
```

Comment:
This is potentially dangerous code.

In the next issue (#222), a person already complains that the component freezes the entire system when it tries to pass an array of 100,000 records through the component.

If I add your code, there is a high risk of total system slowdown at much smaller volumes.

Please try to solve the same problem without re-storing the entire data array in memory.

Reply:
Don't know why i did it this way :P Changed it to only store the last_state in its own variable and not all values.

Reply:
Nevermind. I looked over my code again. It stores the last values for each entity in its own list to use the average of that list for the trend. I added a check to only append to the list if the last value is of instance number (just like your own code). I don't see how this new code is any more unsafe than your own. So if you still think it is unsafe, please explain how i can improve that